src/libs/Navigation/CustomActions.js

```diff
@@ -24,7 +60,7 @@ function pushDrawerRoute(screenName, params, navigationRef) {
 
         if (activeReportID === params.reportID) {
             if (state.type !== 'drawer') {
-                navigationRef.current.dispatch(StackActions.pop());
+                navigateBackToDrawer(navigationRef);
```

Comment:
One thing that sticks out to me is that we are checking `state.type !== 'drawer'` and then again checking in `navigateBackToDrawer()`. So it seems the method has more responsibility than it needs.